docs/strategies/hedging.md

```diff
@@ -440,6 +440,40 @@ sequenceDiagram
     P->>C: Propagates failure (R3)
 ```
 
+## Hedging and resilience context
+
+The hedging strategy supports the concurrent execution and cancellation of multiple hedged actions. To safeguard against concurrency issues when these actions access the same `ResilienceContext`, the hedging strategy ensures that each hedged action has its unique `ResilienceContext`. The hedging strategy distinguishes between two types of contexts:
+
+- **Primary context**: The original resilience context that the hedging strategy receives.
+- **Action context**: A deep copy of the primary context with distinct cancellation token for a single hedged action.
+
+Here's the flow:
+
+- The strategy gets the primary context and creates a snapshot for deep cloning.
+- For each hedged action execution, the hedging strategy makes a deep copy of the original context. The deep copy has its own cancellation token designated for that execution. Note that the first execution (primary) uses the original resilience context, albeit with a cloned set of resilience properties.
+- After the strategy has an accepted result from a hedged action, the resilience context from the action is merged back into the primary context.
+- All ongoing hedged actions are cancelled and discarded. The hedging strategy awaits the propagation of cancellation.
+
+### Merging action context into primary context
+
+After merging, the primary context contains:
+
+- New properties saved during the action context's execution.
+- Existing properties updated during execution.
+
+> [!NOTE]
+> If you delete a property from the action context and this property exists in the primary context, the change is ignored, and the primary context retains the original property.
```

Comment:
As far as I can see you can't delete a property from the context. Only `GetValue`, `TryGetValue` and `Set` are exposed publicly. I've used the term soft-delete intentionally inside the issue to describe a case when you set a proprety value to `null`.

Reply:
You are right, dropped this section and updated the list with this info.